pkg/causality/internal/node.go

```diff
@@ -46,10 +48,11 @@ var (
 // in conflict detection.
 type Node struct {
 	// Immutable fields.
-	id int64
+	id                  int64
+	sortedDedupKeysHash []uint64
 
 	// Called when all dependencies are resolved.
-	OnResolved func(id workerID)
+	TrySendToWorker func(id workerID) bool
```

Comment:
IMHO, `OnResolved` is a better name than `TrySendToWorker` because the node doesn't care about the outside logic.

Reply:
The callback function is named differently to avoid confusion, since There are several functions in node that are named resolved.